packages/catalog-process/src/services/catalogService.ts

```diff
@@ -605,13 +600,11 @@ export async function deleteDocumentLogic({
   assertEServiceExist(eServiceId, eService);
   assertRequesterAllowed(eService.data.producerId, authData.organizationId);
 
-  const descriptor = eService.data.descriptors.find(
-    (d: Descriptor) => d.id === descriptorId
-  );
+  const descriptor = retrieveDescriptor(descriptorId, eService);
```

Comment:
Refactor and minor change.
Previously, the `eServiceDescriptorNotFound` wasn't handled separately in this flow, so that case resulted in a `eServiceDocumentNotFound` error in the next lines. With the change, the two assertions happen at different moments so we could distinguish between each. Indeed, the `retrieveDescriptor` function throws an `eServiceDescriptorNotFound` if it doesn't find the required descriptor.